client/src/features/ProjectPageV2/ProjectPageContent/Settings/ProjectKeywordsFormField.tsx

```diff
@@ -0,0 +1,122 @@
+}
+
+export default function ProjectKeywordsFormField({
+  control,
+  errors,
+  getValues,
+  oldKeywords,
+  setValue,
+}: ProjectKeywordsFormFieldProps) {
+  return (
+    <div>
+      <Label className="form-label" for="project-keyword">
+        Keywords
+      </Label>
+      <div className={cx("input-group", "input-group-sm", "mb-2")}>
+        <Controller
+          control={control}
+          name="keyword"
+          render={({ field }) => (
+            <>
+              <input
+                id="project-keyword"
+                placeholder="Add new keyword"
+                type="string"
+                {...field}
+                className={cx("form-control", errors.keyword && "is-invalid")}
+                data-cy="project-settings-keyword-input"
+                onChange={(e) => {
+                  field.onChange(e);
+                }}
+                onKeyDown={(e) => {
+                  if (e.key === "Enter" && field.value) {
+                    e.preventDefault();
+                    const newValue = field.value.trim();
+                    const currentKeywords = getValues("keywords");
+                    if (!currentKeywords.includes(newValue)) {
+                      const newKeywords = [...currentKeywords, newValue];
+                      setValue("keywords", newKeywords);
+                    }
+                    setValue("keyword", "");
+                  }
+                }}
+              />
+              <Button
+                color={field.value ? "primary" : "outline-primary"}
+                disabled={!field.value}
+                data-cy="project-settings-keyword-button"
+                onClick={() => {
+                  if (field.value) {
+                    const newValue = field.value.trim();
+                    const currentKeywords = getValues("keywords");
+                    if (!currentKeywords.includes(newValue)) {
+                      const newKeywords = [...currentKeywords, newValue];
+                      setValue("keywords", newKeywords);
+                    }
+                    setValue("keyword", "");
+                  }
+                }}
+                type="button"
+              >
+                <PlusLg className={cx("bi", "me-1")} />
+                Add
+              </Button>
+            </>
+          )}
+        />
+      </div>
+      <Controller
+        name="keywords"
+        control={control}
+        render={({ field }) => (
+          <>
+            {field.value && field.value.length > 0 && (
+              <KeywordContainer data-cy="project-settings-keywords">
+                {getValues("keywords")
+                  .sort((a, b) => a.localeCompare(b))
+                  .map((keyword, index) => (
+                    <KeywordBadge
+                      data-cy="project-settings-keyword"
+                      key={index}
+                      highlighted={!oldKeywords?.includes(keyword)}
```

Comment:
I think it should be equivalent to use `formState` (from the `render` function) here:
```suggestion
                      highlighted={!(formState.defaultValues?.keywords ?? []).includes(keyword)}
```